.github/workflows/python-tests.yml

```diff
@@ -0,0 +1,28 @@
+name: Run Python Tests
+on:
```

Comment:
This will not perform tests on each push, which means contributors will not know about any failing tests until they create a PR or trigger the action manually. I do not know how desired this behavior is, but I wanted to note my observations. For example, in code linting workflow we covered these situations.

Reply:
Good point. Had that originally and then decided to simplify but you're right - it's a good change so will put back.

Reply:
Fixed!